lib/src/style/style.dart

```diff
@@ -37,16 +37,16 @@ class SpriteStyle {
 class StyleReader {
   final String uri;
   final String? apiKey;
+  final Map<String, dynamic>? styleData;
   final Logger logger;
 
-  StyleReader({required this.uri, this.apiKey, Logger? logger})
+  StyleReader({required this.uri, this.apiKey, this.styleData, Logger? logger})
       : logger = logger ?? const Logger.noop();
 
   Future<Style> read() async {
     final uriMapper = StyleUriMapper(key: apiKey);
     final url = uriMapper.map(uri);
-    final styleText = await _httpGet(url);
-    final style = await compute(jsonDecode, styleText);
+    final style = styleData ?? await _fetchStyle(url);
```

Comment:
It's a good idea to provide a way to read a style from local assets, but making this style reader do the work makes it more complicated than it needs to be. For example, there are multiple points below where we reference the URI.

Instead of modifying this style reader, would it be better to provide a separate one? Alternatively, you can instantiate a `Style` directly in your app by calling the constructor.